src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+	}
+
+	protected String getClientCDPPath() {
+
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings != null) {
+			String path = settings.getProperty(CUSTOM_SUBMIT_SAMPLE_CDP_PATH);
+			if (path != null) {
+				return path;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Reset fields back to default values
+	 * 
+	 * @param usingGui
+	 */
+	protected void resetFields(boolean isUsingGui) {
+	    
+	    
+	    selectedJudgingTypeIndexes = new int[0];
+	    selectedProblemsIndexes = new int[0];
+	    selectedLanguagesIndexes = new int[0];
+
+		checkBoxSubmitYesSamples.setSelected(true);
+		checkBoxSubmitFailingSamples.setSelected(true);
+		
+		checkBoxLanguage.setSelected(true);
+        checkBoxProblems.setSelected(true);
+        
+        submitAllJudgingTypesRadioButton.setSelected(true);
+        submitAllProblemsRadioButton.setSelected(true);
+        submitAllLanguagesRadioButton.setSelected(true);
+
+		String cdpPath = getContest().getContestInformation().getJudgeCDPBasePath();
+
+		String clientPath = getClientCDPPath();
+
+		if (clientPath != null && isUsingGui) {
+			int result = FrameUtilities.yesNoCancelDialog(this, "Overwrite locally saved CDP path with this (default) " + cdpPath,
+					"Replace CDP Path?");
+			if (result == JOptionPane.NO_OPTION) {
+				cdpPath = clientPath;
+			}
+		}
+
+		cdpTextField.setText(cdpPath);
+
+//		getController().updateClientSettings(settings);;
+
+	}
+
+	/**
+	 * Submit sample solutions.
+	 */
+	protected void submitSelectedJudgesSolutions() {
+	    showMessage("");
+
+	    String warningMessage = verifyCDP (cdpTextField.getText());
+	    if (warningMessage != null) {
+	        // let the user know that the CDP selected may not work
+	        FrameUtilities.showMessage(this, "Trying to use an invalid CDP?", warningMessage);
+	    }
+
+        List<File> allFiles = ListUtilities.getAllJudgeSampleSubmissionFilenamesFromCDP(getContest(), cdpTextField.getText());
+
+        if (allFiles.size() == 0) {
+            FrameUtilities.showMessage(this, "No Runs", "No Runs found under " + cdpTextField.getText());
+            return;
+        }
+
+        boolean submitYesSamples = checkBoxSubmitYesSamples.isSelected();
+        boolean submitNoSamples = checkBoxSubmitFailingSamples.isSelected();
+
+        if (!submitYesSamples && !submitNoSamples) {
+            FrameUtilities.showMessage(this, "No Runs", "Select either Yes or Failed runs");
+            return;
+        }
+
+        List<File> files = new ArrayList<File>();
+
+        files = QuickSubmitter.filterRuns(allFiles, submitYesSamples, submitNoSamples);
+
+        if (submitNoSamples && submitSelectedJudgingTypeRadioButton.isSelected()) {
+            if (submitSelectedJudgingTypeRadioButton.isSelected()) {
+                // if they selected the radio button for selected, they should select at least
+                if (selectedJudgingTypeIndexes == null || selectedJudgingTypeIndexes.length == 0) {
```

Comment:
Every time this condition is hit, a BLANK message is written to the output log/textbox.  That seems useless.